### PR TITLE
Python: Delete filter queries, code duplication library, and precision tag from metric queries

### PR DESCRIPTION
**Pull request by RasmusWL**
Port of the [Java PR](https://github.com/github/codeql/pull/5525), which is slightly different from the [C++](https://github.com/github/codeql/pull/5515) and [C#](https://github.com/github/codeql/pull/5524) PRs.

### CODE REVIEW

**Comment by aibaars** [comment]
@RasmusWL Having the old queries in the experimental folder might cause problems for customers that want to run real experimental queries with CodeScanning. I doubt any customers were doing that, otherwise you'd have had some complaints about missing `@id` when trying to interpret query results to SARIF format. So it's not an immediate problem, however, now python is part of the bounty program the experimental folder may soon contain lots of new interesting queries that folk want to try.

**Comment by RasmusWL** [comment]
Thanks for the extra insight @aibaars. I think we'll work on removing them soon-ish, but for this PR i just wanted to fix up things so we can move on :rocket:

**Review by asgerf — APPROVED** [review]
LGTM but as I'm not part of `codeql-python` I'm not sure it's enough make the PR mergable.

File: python/ql/src/experimental/Security-old-dataflow/CWE-094/CodeInjection.ql [hunk 1]
```diff
@@ -3,6 +3,8 @@
  * @description Interpreting unsanitized user input as code allows a malicious user arbitrary
```
Comment (calumgrant) [inline python/ql/src/experimental/Security-old-dataflow/CWE-094/CodeInjection.ql:3]: Put "OLD QUERY: " in front of the name.

File: python/ql/src/external/DuplicateBlock.ql [hunk 2]
```diff
@@ -16,19 +16,7 @@
...
-from DuplicateBlock d, DuplicateBlock other
-where
-  d.sourceLines() > 10 and
-  other.getEquivalenceClass() = d.getEquivalenceClass() and
-  sorted_by_location(other, d)
-select d,
-  "Duplicate code: " + d.sourceLines() + " lines are duplicated at " +
-    other.sourceFile().getShortName() + ":" + other.sourceStartLine().toString()
+from BasicBlock d
+where none()
```
Comment (calumgrant) [inline python/ql/src/external/DuplicateBlock.ql:21]: I'm not sure why we've half-removed the query. What is the purpose of removing CodeDuplication.qll, but keeping all the queries that depend on it? I think we should either remove these queries completely, or keep them working.

Comment (calumgrant) [reply python/ql/src/external/DuplicateBlock.ql:21]: Is there a technical reason to remove CodeDuplication.qll at this time?

File: python/ql/src/experimental/Security-old-dataflow/CWE-601/UrlRedirect.ql [hunk 3]
```diff
@@ -3,6 +3,8 @@
  * @description URL redirection based on unvalidated user input
  *              may cause redirection to malicious web sites.
  * @kind path-problem
+ * @problem.severity error
+ * @id py/old/url-redirection
```
Comment (calumgrant) [inline python/ql/src/experimental/Security-old-dataflow/CWE-601/UrlRedirect.ql:7]: To be honest I'd be happier to just delete these queries, as they are becoming a maintenance burden with no benefit.

**Review by calumgrant — COMMENTED** [review]
Just a few questions, but overall I think this seems very reasonable and I'm happy to merge this.

File: python/ql/src/experimental/Security-old-dataflow/CWE-601/UrlRedirect.ql (hunk 3 above)
Comment (RasmusWL) [reply python/ql/src/experimental/Security-old-dataflow/CWE-601/UrlRedirect.ql:7]: While I agree that we should do this soon, I don't really make that decision a part of this PR (as explained above)

File: python/ql/src/external/DuplicateBlock.ql (hunk 2 above)
Comment (RasmusWL) [reply python/ql/src/external/DuplicateBlock.ql:21]: This PR is simply following what Java did, which is based on internal discussion on slack. The main idea is to not break any query-suites that still reference the code duplication queries.

File: python/ql/src/experimental/Security-old-dataflow/CWE-094/CodeInjection.ql (hunk 1 above)
Comment (RasmusWL) [reply python/ql/src/experimental/Security-old-dataflow/CWE-094/CodeInjection.ql:3]:
```suggestion
 * @description OLD QUERY: Interpreting unsanitized user input as code allows a malicious user arbitrary
```

File: python/ql/src/external/DuplicateBlock.ql (hunk 2 above)
Comment (calumgrant) [reply python/ql/src/external/DuplicateBlock.ql:21]: Ah I see, if the data isn't in the database then it's safe to delete this file.

**Review by calumgrant — APPROVED** [review]

**Pull request merged**